src/Module/Race.cpp

```diff
@@ -6,6 +6,90 @@
 
 #include <Vehicle/TireControlE.h>
 
+void Race::Initialize(VehiclePtr vehicle) {
+    sf::Vector2f pos = map_->GetStartPosition();
+    vehicle_ = vehicle;
+    // Prepare map
+    vehicle_->Initialize(world_, (int) pos.x, (int) pos.y);
+
+    checkPointManager_->Reset();
+    hud_->Initialize(vehicle_, checkPointManager_);
+    map_->AlignCameraViewSize(*window_);
+    map_->SetCameraViewPosition(vehicle_->GetPosition());
+    world_->SetContactListener(&contactListener_);
```

Comment:
ContactListener jest naszą małą klasą i raczej nie byłoby problemu trzymać jąw smartptr i brać od niej plain ptr wrzucając ją do worlda. Wtedy aby być pewnym że contact litener będzie żył conajmniej tyle co world dobrze byłoby też trzymać worlda w sm ptr. Jeśli chcesz to możesz to zostawić.